src/lib/childrenUtils.js

```diff
@@ -16,3 +16,14 @@ export const someByType = (children, type) => _.some(Children.toArray(children),
  * @returns {undefined|Object}
  */
 export const findByType = (children, type) => _.find(Children.toArray(children), { type })
+
+/**
+ * Tests if any children.
+ * @param {Object} children The children prop of a component.
+ * @returns {Boolean}
+ */
+export const isNil = (children) => {
+  if (children == null) return true // eslint-disable-line
+  if (Array.isArray(children) && children.length == 0) return true // eslint-disable-line
```

Comment:
Sure, the `==` simply coerces null/undefined comparisons to return true.  We can just explicitly list those. `[].length` always returns a number to a triple equal there should be safe.  Something like:

```js
export const isNil = children => {
  return children === null
    || children === undefined
    || Array.isArray(children) && children.length === 0
}
```